lib/vsc/install/shared_setup.py

```diff
@@ -623,20 +791,118 @@ def sanitize(name):
     return ",".join([sanitize(r) for r in name])
 
 
-def parse_target(target):
+def get_md5sum(filename):
+    """Use this function to compute the md5sum in the KNOWN_LICENSES hash"""
+    return hashlib.md5(open(filename).read()).hexdigest()
+
+def get_license(license=None):
+    """
+    Determine the license of this project based on LICENSE file
+
+    license argument is the license file to check. if none rpovided, the project LICENSE is used
+    """
+    # LICENSE is required and enforced
+    if license is None:
+        license = os.path.join(REPO_BASE_DIR, LICENSE)
+    if not os.path.exists(license):
+        raise Exception('LICENSE is missing (was looking for %s)' % license)
+
+    license_md5 = get_md5sum(license)
+    log.info('found license %s with md5sum %s' % (license, license_md5))
+    found_lic = False
+    for lic_short, data in KNOWN_LICENSES.items():
+        if license_md5 != data[0]:
+            continue
+
+        found_lic = True
+        break
+
+    if not found_lic:
+        raise Exception('UNKONWN LICENSE %s provided. Should be fixed or added to vsc-install' % license)
+
+    log.info("Found license name %s and classifier %s" , lic_short, data[1])
+    return lic_short, data[1]
+
+
+def parse_target(target, urltemplate=None):
     """
     Add some fields
+        get name / ur/ download url from project
+        set url / download_url from template
+
+        vsc_description: set the description and long_description from the README
+        vsc_scripts: generate scripts from bin content
 
     Remove sdist vsc class with '"vsc_sdist": False' in target
     """
     new_target = {}
     new_target.update(SHARED_TARGET)
 
+    if not 'name' in target:
+        log.info('No name defined, trying to determine it')
+        # sets name / url and download_url
+        target.update(get_name_url())
+
+    # prepare classifiers
+    classifiers = new_target.setdefault('classifiers', [])
+
+    if urltemplate:
+        new_target['url'] = urltemplate % target
+        if 'github' in urltemplate:
+            new_target['download_url'] = "%s/tarball/master" % new_target['url']
```

Comment:
master will surely be wrong, try `%s/archive/%(name)s-V%(version)s.tar.gz`
and make sure the release script does this release on github and pipy